src/Sylius/Bundle/PromotionBundle/Doctrine/ORM/PromotionCouponRepository.php

```diff
@@ -45,6 +45,30 @@ public function countByCodeLength(int $codeLength): int
         ;
     }
 
+    public function countByCodeLengthWithPrefixAndSuffix(
+        int $codeLength,
+        ?string $prefix = null,
+        ?string $suffix = null
+    ): int {
+        if ($prefix !== null) {
+            $codeLength += strlen($prefix);
+        }
+        if ($suffix !== null) {
+            $codeLength += strlen($suffix);
+        }
+        $codeTemplate = $prefix . '%' . $suffix;
+
+        return (int) $this->createQueryBuilder('o')
+            ->select('COUNT(o.id)')
+            ->andWhere('LENGTH(o.code) = :codeLength')
+            ->andWhere('o.code LIKE :prefixAndSuffix')
```

Comment:
Personal preference, it brings more context into this query.

```suggestion
            ->andWhere('o.code LIKE :codeTemplate')
```